iceoryx_hoofs/include/iceoryx_hoofs/internal/posix_wrapper/shared_memory_object/shared_memory.hpp

```diff
@@ -36,12 +36,15 @@ enum class AccessMode : uint64_t
 };
 static constexpr const char* ACCESS_MODE_STRING[] = {"AccessMode::READ_ONLY", "AccessMode::READ_WRITE"};
 
-enum class OwnerShip : uint64_t
+enum class Policy : uint64_t
```

Comment:
I consulted @ithier and he suggested `Policy`. @elBoberido would you accept this? `OpenOption::OPEN` sounds just weird and `Mode` is already taken.

Reply:
Well, I'm just pointing out you are inventing new terminology. The STL has [openmode](https://en.cppreference.com/w/cpp/io/basic_filebuf/open), Boost has [open_mode](https://www.boost.org/doc/libs/1_76_0/boost/log/utility/open_mode.hpp) for logging, Qt has [OpenMode](https://doc.qt.io/qt-5/qiodevice.html#OpenModeFlag-enum), Rust has [OpenOptions](https://doc.rust-lang.org/std/fs/struct.OpenOptions.html) and Python has just [mode](https://docs.python.org/3/library/functions.html#open). They mostly also incorporate `AccessMode` into that parameter.

By just having the signature `bool open(const AccessMode accessMode, const Policy policy, const mode_t permissions, const uint64_t size) noexcept;` I would have no clue what `Policy policy` means, contrary to the other parameter. Although, it wouldn't be much different with `Ownership`.

But this is not only about opening files, but specifically about shared memory which has certain implications on ownership for cleanup. Let's have a look at boost [shared_memory_object](https://www.boost.org/doc/libs/1_76_0/doc/html/boost/interprocess/shared_memory_object.html) and Qt's [QSharedMemory](https://doc.qt.io/qt-5/qsharedmemory.html). Boost uses different ctors and Qt uses different methods. I kind of like the approach of Qt with `attach` and `create` which could also be used to indicate ownership, although I think they are using some memory to store meta information like this. For us, `attach` and `create` would clearly show the ownership situation. With `create` maybe having an `OnExisting` enum with `PURGE` and `ERROR`. The current `OPEN_OR_CREATE` would done by `create` followed by `attach`, maybe in an `attach_or_create` convenience method. Not sure how usable such an API would be and definitely nothing for this PR.

To make a long story short, please consider to not introduce new terminology ... oh and it could be `OpenOption::OPEN_ONLY` or `OpenMode::OPEN_ONLY` like boost does with `open_mode` ;)

Reply:
After the first paragraph you already convinced me. So you wrote the next three paragraphs for nothing muahahaha ;)

Reply:
Doh :smile: